models/experimental/functional_unet/tt/unet_shallow_ttnn.py

```diff
@@ -395,32 +346,10 @@ def __init__(self, parameters: ParameterDict, device, mesh_mapper=None) -> None:
         )
 
         self.bnc = UNetConv2D(parameters.bnc, parameters.bnb, device, cache=self.conv_cache, mesh_mapper=mesh_mapper)
+        self.bnc.conv_config.reshard_if_not_optimal = True
```

Comment:
Pass `should_reshard` to `UNetConv2D` instead?

Reply:
sure